project/Dependencies.scala

```diff
@@ -6,39 +6,268 @@ object Dependencies {
+
+    ) ++ crossLibraryFamily("com.typesafe.play", PlayVersion)(
+      "play", "play-datacommons", "play-functional", "play-iteratees", "play-java", "play-jdbc", "play-jdbc-api",
+      "play-json", "play-netty-server", "play-server", "play-streams", "play-ws"
+
+    ) ++ libraryFamily("ch.qos.logback", "1.1.3")(
+      "logback-classic", "logback-core"
+
+    ) ++ libraryFamily("io.netty", NettyVersion)(
+      "netty-buffer", "netty-codec", "netty-codec-http", "netty-common", "netty-handler", "netty-transport",
+      "netty-transport-native-epoll"
+
+    ) ++ libraryFamily("org.apache.logging.log4j", "2.7")(
+      "log4j-api", "log4j-core", "log4j-slf4j-impl"
+
+    ) ++ libraryFamily("org.asynchttpclient", "2.0.11")(
+      "async-http-client", "netty-codec-dns", "netty-resolver", "netty-resolver-dns"
+
+    ) ++ libraryFamily("org.ow2.asm", "5.0.3")(
+      "asm", "asm-analysis", "asm-commons", "asm-tree", "asm-util"
+
+    ) ++ libraryFamily("org.scala-lang", scalaVersion)(
+      "scala-library", "scala-reflect"
+
+    ) ++ libraryFamily("org.slf4j", "1.7.21")(
+      "jcl-over-slf4j", "jul-to-slf4j", "log4j-over-slf4j", "slf4j-api"
+    )
+  }
+
+  // These dependencies are used by JPA to test, but we don't want to export them as part of our regular whitelist,
+  // so we maintain it separately.
+  val JpaTestWhitelist = Seq(
+    "antlr" % "antlr" % "2.7.7",
+    "dom4j" % "dom4j" % "1.6.1",
+    "javax.annotation" % "jsr250-api" % "1.0",
+    "javax.el" % "el-api" % "2.2",
+    "javax.enterprise" % "cdi-api" % "1.1",
+    "org.apache.geronimo.specs" % "geronimo-jta_1.1_spec" % "1.1.1",
+    "org.hibernate" % "hibernate-core" % "5.2.5.Final",
+    "org.hibernate.common" % "hibernate-commons-annotations" % "5.0.1.Final",
+    "org.jboss" % "jandex" % "2.0.3.Final",
+    "org.jboss.spec.javax.interceptor" % "jboss-interceptors-api_1.1_spec" % "1.0.0.Beta1"
+  )
+
+  // These dependencies are used by the Kafka tests, but we don't want to export them as part of our regular
+  // whitelist, so we maintain it separately.
+  val KafkaTestWhitelist = Seq(
+    "com.101tec" % "zkclient" % "0.8",
+    "com.yammer.metrics" % "metrics-core" % "2.2.0",
+    "jline" % "jline" % "0.9.94",
+    "log4j" % "log4j" % "1.2.17",
+    "net.sf.jopt-simple" % "jopt-simple" % "4.9",
+    "org.apache.commons" % "commons-math" % "2.2",
+    "org.apache.curator" % "curator-client" % "2.10.0",
+    "org.apache.curator" % "curator-framework" % "2.10.0",
+    "org.apache.curator" % "curator-test" % "2.10.0",
+    "org.apache.kafka" %% "kafka" % "0.10.0.1",
+    "org.apache.zookeeper" % "zookeeper" % "3.4.6"
+  )
 
+  private def crossLibraryFamily(groupId: String, version: String)(artifactIds: String*) = {
+    artifactIds.map(aid => groupId %% aid % version)
+  }
+
+  private def libraryFamily(groupId: String, version: String)(artifactIds: String*) = {
+    artifactIds.map(aid => groupId % aid % version)
+  }
```

Comment:
There's no hard criteria, library families are just a convenience to avoid excess boilerplate in this already very long file.